src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3485,6 +3486,46 @@ internal void SymbolDeclaredEvent(Symbol symbol)
             EventQueue?.TryEnqueue(new SymbolDeclaredCompilationEvent(this, symbol.GetPublicSymbol()));
         }
 
+        internal override void SerializePdbEmbeddedCompilationOptions(BlobBuilder builder)
+        {
+            WriteValue("checked", Options.CheckOverflow.ToString());
+            WriteValue("nullable", Options.NullableContextOptions.ToString());
+            WriteValue("unsafe", Options.AllowUnsafe.ToString());
+            WriteValue("language-version", LanguageVersion.ToString());
+
+            var optimization = Options.OptimizationLevel == OptimizationLevel.Debug
+                ? (Options.DebugPlusMode ? "debug+" : "debug")
+                : "release";
```

Comment:
1. @tmat said using ToString() on enum was a nono :P 
2.  Not all values for this are in the enum yet. #44809

Reply:
`LanguageVersion` is an `enum` and you use `ToString` three lines above so this logic is not consistently applied to the change.

Reply:
The argument for not using `ToString()` is performance based, since it uses reflection, but I think it's negligible here. The main point is that not all values needed are covered by the enum

Reply:
> The main point is that not all values needed are covered by the enum

This still doesn't make sense to me. We're essentially encoding two options as one option here. Why not just encode both options individually. This is the simplest approach to the problem.

Reply:
They _should_ be in the same enum though, they're just not. Part of the goal of #44809 is to add `debug+` to the OptimizationLevel enum

Reply:
As a part of #44809 we'd have to do exactly what I'm proposing here: serialize the `enum` directly. Seems simpler to do that now, add in the temporary option and then just remove that when the other bug is fixed. That would mean the contents of the field in the PDB are consistent through the lifetime of the feature. Where if we go with the current approach PDB files created in one time period will have one format for the field (the generated string) and others will have the serialized `enum`

Reply:
If we keep it like this, it's not a breaking change at all whether it's in the enum or not. The pdb information still accurately represents the contract. 

There are two other enums we use in serialization that we should consider thinking about as well: language version and nullability context. 

For language version I'd like to switch to use the display format, "7.1" for example, and add a language field as well. 

For nullability I think it's fine that it directly is serialized, and we can list all of the values in the documentation of the pdb contract.

Reply:
I think we've come to an agreement on what to do here from talking offline. Marking as resolved